azure-pipelines-external.yml

```diff
@@ -0,0 +1,150 @@
+        displayName: 'Run Code Analysis - Begin Step (Pull Request)'
+        condition: eq(variables['Build.Reason'], 'PullRequest')
+        inputs:
+          SonarCloud: 'SonarCloud'
+          organization: 'sonarsource'
+          scannerMode: 'MSBuild'
+          projectKey: 'sonaranalyzer-dotnet'
+          projectName: 'SonarAnalyzer .Net'
+          projectVersion: '$(SONAR_PROJECT_VERSION)'
+          extraProperties: |
+            sonar.analysis.buildNumber=$(Build.BuildId)
+            sonar.analysis.pipeline=$(Build.BuildId)
+            sonar.analysis.sha1=$(System.PullRequest.SourceCommitId)
+            sonar.analysis.prNumber=$(System.PullRequest.PullRequestNumber)
+            sonar.analysis.repository=$(Build.Repository.ID)
+
+      - task: SonarCloudPrepare@1
+        displayName: 'Run Code Analysis - Begin Step (master or branch)'
+        condition: ne(variables['Build.Reason'], 'PullRequest')
+        inputs:
+          SonarCloud: 'SonarCloud'
+          organization: 'sonarsource'
+          scannerMode: 'MSBuild'
+          projectKey: 'sonaranalyzer-dotnet'
+          projectName: 'SonarAnalyzer .Net'
+          projectVersion: '$(SONAR_PROJECT_VERSION)'
+          extraProperties: |
+            sonar.analysis.buildNumber=$(Build.BuildId)
+            sonar.analysis.pipeline=$(Build.BuildId)
+            sonar.analysis.sha1=$(Build.SourceVersion)
+            sonar.analysis.repository=$(Build.Repository.ID)
+
+      - task: VSBuild@1
+        displayName: 'Build SonarAnalyzer solution'
+        inputs:
+          solution: '$(solution)'
+          platform: '$(buildPlatform)'
+          configuration: '$(buildConfiguration)'
+          msbuildArgs: '/p:DeployExtension=false /p:Sha1=$(Build.SourceVersion) /p:BuildNumber=$(Build.BuildId)'
+
+      - task: PublishPipelineArtifact@1
+        displayName: 'Publish analyzer binaries as pipeline artifact'
+        inputs:
+          path: sonaranalyzer-dotnet/its/binaries/
+          artifact: Binaries
+
+      - task: PublishPipelineArtifact@1
+        displayName: 'Publish rule descriptor binaries as pipeline artifact'
+        inputs:
+          path: sonaranalyzer-dotnet\src\SonarAnalyzer.RuleDescriptorGenerator\bin
+          artifact: RuleDescriptorBin
+
+      - task: VSTest@2
+        displayName: "Run SonarAnalyzer unit tests"
+        inputs:
+          testAssemblyVer2: |
+            **\$(BuildConfiguration)\**\*.UnitTest.dll
+            !**\obj\**
+          runInParallel: true
+          runTestsInIsolation: true
+          codeCoverageEnabled: true
+          platform: '$(buildPlatform)'
+          configuration: '$(buildConfiguration)'
+          rerunFailedTests: false
+
+      - task: SonarCloudAnalyze@1
+        displayName: 'Run Code Analysis - End Step'
+
+      - task: SonarCloudPublish@1
+        displayName: 'Run Code Analysis - Publish QG'
+        inputs:
+          pollingTimeoutSec: '300'
+
+    - job: runIntegrationTestsJob
```

Comment:
In this case we have just 2 jobs which need to run in a specific order since `runIntegrationTestsJob` depends on `runUnitTestsJob`. I think we should refactor this to have only one job and remove the tasks PublishPipelineArtifact and DownloadPipelineArtifact. 

I think that overall it will speed up the build and make things simpler.